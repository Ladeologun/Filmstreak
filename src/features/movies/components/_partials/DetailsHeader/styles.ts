import { StyleSheet, Platform } from "react-native";

const styles = StyleSheet.create({
    barcontainer:{
        flexDirection:"row",
        justifyContent:"space-between",
        alignItems:"center",
        marginTop:Platform.OS ==="ios"? 40: 20,
        paddingHorizontal:24,
    },
    backarrow:{
        justifyContent:"center",
        alignItems:"center",
        width:50,
        height:50,
        borderRadius:20,
        backgroundColor:'rgba(0, 0, 0, 0.4)'
    }
})

export default styles;