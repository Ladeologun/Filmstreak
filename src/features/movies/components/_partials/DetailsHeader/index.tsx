import {Image, Text, TextStyle, TouchableOpacity, View, ViewStyle} from 'react-native';
import styles, { BackButton } from './styles';
import {LEFT_ARROW, WISHLIST_ICON} from '~movies/constants';
import {useNavigation} from '@react-navigation/native';
import {Walkthrough} from '~components';
import {useEffect, useRef} from 'react';
import {COLORS} from '~styles';
import {WalkthroughHandle} from '~components';
import {ROUTES} from '~navigation/routes';
import {NavigationTypes} from '~movies/types';

interface Iprops {
   containerStyle?: ViewStyle;
   hasTitle?: boolean;
   title?: string;
   titleStyle?: TextStyle;
   hasRightCTA?: boolean;
   showWalkthrough?: boolean;
   handleWalkthroughClose?: () => void;
}

const DetailsHeader: React.FC<Iprops> = ({
   containerStyle,
   hasRightCTA,
   hasTitle,
   titleStyle,
   title,
   showWalkthrough,
   handleWalkthroughClose,
}) => {
   const navigation = useNavigation<NavigationTypes>();
   const wishlistWalkthroughRef = useRef<WalkthroughHandle | null>(null);

   useEffect(() => {
      if (hasRightCTA && showWalkthrough) {
         wishlistWalkthroughRef?.current?.open();
      }
   }, [hasRightCTA, showWalkthrough]);

   return (
      <View style={[styles.barcontainer, containerStyle]}>
         <BackButton
            style={styles.backarrow}
            onPress={() => {
               navigation.goBack();
            }}
         >
            <Image source={LEFT_ARROW} style={styles.leftIconStyle} />
         </BackButton>

         {!!hasTitle && <Text style={[{color: COLORS.WHITE}, titleStyle]}>{title}</Text>}

         {!!hasRightCTA ? (
            <Walkthrough
               ref={wishlistWalkthroughRef}
               onNext={() => {
                  wishlistWalkthroughRef?.current?.close();
                  handleWalkthroughClose?.();
               }}
               hidePrevious
               contentTitle="Wishlist"
               nextBtnText="Done"
               contentDesc="Click this icon to view movies in your wishlist"
               contentContainerStyle={{alignSelf: 'flex-end', marginRight: 15}}
               triangleStyle={{right: 14}}
               backgroundColor={COLORS.PRIMARY}
            >
               <TouchableOpacity
                  style={styles.backarrow}
                  onPress={() => {
                     navigation.navigate(ROUTES.MOVIE_WISHLISTS);
                  }}
               >
                  <Image source={WISHLIST_ICON} style={{width: 25, height: 25}} />
               </TouchableOpacity>
            </Walkthrough>
         ) : (
            <View />
         )}
      </View>
   );
};

export default DetailsHeader;
