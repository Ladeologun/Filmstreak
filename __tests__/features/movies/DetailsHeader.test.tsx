import renderer from 'react-test-renderer';
import {test,expect} from '@jest/globals';
import DetailsHeader from '~movies/components/_partials/DetailsHeader';


test('renders correctly', () => {
    const tree = renderer.create(<DetailsHeader hasTitle={true} title='filmstreak' hasRightCTA={true}/>).toJSON();
    expect(tree).toMatchSnapshot();
});
