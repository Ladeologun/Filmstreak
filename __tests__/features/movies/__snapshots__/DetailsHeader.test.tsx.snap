// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 40,
        "paddingHorizontal": 24,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.4)",
        "borderBottomLeftRadius": 20,
        "borderBottomRightRadius": 20,
        "borderRadius": 20,
        "borderTopLeftRadius": 20,
        "borderTopRightRadius": 20,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "width": 50,
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/assets/drawables/left-arrow.png",
        }
      }
      style={
        {
          "height": 20,
          "tintColor": "#ffffff",
          "width": 20,
        }
      }
    />
  </View>
  <Text
    style={
      [
        {
          "color": "#ffffff",
        },
        undefined,
      ]
    }
  >
    filmstreak
  </Text>
  <View>
    <View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 0, 0, 0.4)",
            "borderRadius": 20,
            "height": 50,
            "justifyContent": "center",
            "opacity": 1,
            "width": 50,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../src/assets/drawables/wish-list.png",
            }
          }
          style={
            {
              "height": 25,
              "width": 25,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
